src/m365/teams/commands/chat/chat-member-add.spec.ts

```diff
@@ -0,0 +1,177 @@
+    assert(postStub.called);
+  });
+
+  it('adds the member by specifying the userId with chat history from a certain date', async () => {
+    const postStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/chats/${chatId}/members`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { chatId: chatId, userId: userId, visibleHistoryStartDateTime: '2019-04-18T23:51:43.255Z' } });
+    assert(postStub.called);
+  });
+
+  it('fails validation if the chatId is not valid chatId', async () => {
+    const actual = await command.validate({ options: { chatId: 'invalid', userId: userId } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if the userId is not valid guid', async () => {
+    const actual = await command.validate({ options: { chatId: chatId, userId: 'invalid' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if the userName is not valid UPN', async () => {
+    const actual = await command.validate({ options: { chatId: chatId, userName: 'invalid' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if the userId is a valid GUID', async () => {
+    const actual = await command.validate({ options: { chatId: chatId, userId: userId } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation if the userName is a valid UPN', async () => {
+    const actual = await command.validate({ options: { chatId: chatId, userName: userPrincipalName } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('fails validation if the role is not valid role', async () => {
+    const actual = await command.validate({ options: { chatId: chatId, userId: userId, role: 'invalid' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if the role is a valid role', async () => {
+    const actual = await command.validate({ options: { chatId: chatId, userId: userId, role: 'guest' } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('fails validation if the visibleHistoryStartDateTime is not valid date', async () => {
+    const actual = await command.validate({ options: { chatId: chatId, userId: userId, visibleHistoryStartDateTime: 'invalid' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if the visibleHistoryStartDateTime is a valid date', async () => {
+    const actual = await command.validate({ options: { chatId: chatId, userId: userId, visibleHistoryStartDateTime: '2019-04-18T23:51:43.255Z' } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('correctly handles API OData error', async () => {
+    const error = {
+      error: {
+        code: 'generalException',
+        message: `The member can't be added to the team`
```

Comment:
Let's use single quotes here.

Reply:
We can't since there is a single quote here with "can't"

Reply:
While reviewing it again, I noticed why you used backticks, because of the `can't`. In this case, backticks were fine. Sorry! (single quotes are fine as well so you don't have to revert it)